### DISCUSSION TITLE
Maximum size of a image pushed to project.

### DISCUSSION THREAD

**Discussion by prashi2202** [goharbor/harbor#15485] [Q&A]
Hi All,

We are testing harbor setup on EKS cluster and we need to know should be the resource size to push a large image (75GB) into a given project within harbor instance.

Current configuration

Cluster EKS:
$ kubectl get pods --kubeconfig kubeconfig-readonly.yaml
NAME                                   READY   STATUS    RESTARTS   AGE
harbor-chartmuseum-7c8f9cbb55-46mn8    1/1     Running   0          15h
harbor-core-5d4f6ff85-mnzxv            1/1     Running   0          15h
harbor-jobservice-68d965b75b-b9sm7     1/1     Running   0          13h
harbor-jobservice-68d965b75b-ssjxb     1/1     Running   1          15h
harbor-notary-server-f5798db99-rsnht   1/1     Running   0          15h
harbor-notary-signer-6d4b6768f-9g778   1/1     Running   0          15h
harbor-portal-5647cf8d45-gcclj         1/1     Running   0          15h
harbor-redis-0                         1/1     Running   0          15h
harbor-registry-66c88cb8f9-hnsq7       2/2     Running   0          15h
harbor-registry-676b988d75-zd8kg       2/2     Running   0          15h
harbor-trivy-0                         1/1     Running   0          15h

Registry resource size:

vas:harbor-registry-676b988d75-zd8kg
  registry:2Gi
  registryctl:2Gi

vas:harbor-registry-676b988d75-zd8kg
  registry:100m
  registryctl:100m

**Comment by dioguerra** [comment]
Did you try to upload it already?

**Reply by prashi2202** [reply]
Yes i did and the pods failed with memory error.

Error creating: pods "harbor-registry-66c88cb8f9-c9zwk" is forbidden: exceeded quota: resource-quota-xxx, requested: limits.memory=2Gi,requests.memory=2Gi, used: limits.memory=7296Mi,requests.memory=7296Mi, limited: limits.memory=8Gi,requests.memory=8Gi

**Comment by dioguerra** [comment]
This happened to me because nginx stores the transient pod in memory. I solved this issue by bypassing the nginx with

```
                        ingress.kubernetes.io/proxy-body-size: "0"
                        nginx.ingress.kubernetes.io/proxy-body-size: "0"
```

Maybe you have the same problem. After this i had no issues uploading big images, with much less memory actually

Your original error tho, is that you are exceeding the quota assigned to your NS, it seems. you have a limit of 2GB max that you can use for ALL your pods, and in this pod you are asking for 4 times more memory. So.... start with less and within your quota limits

**Reply by prashi2202** [reply]
Thanks for the inputs dioguerra, the current config in my values.yaml is as below..

kubernetes.io/ingress.class: nginx
nginx.ingress.kubernetes.io/ssl-redirect: "true"
nginx.ingress.kubernetes.io/proxy-body-size: "0"

Also the other pods are configured use 128Mi and 50m CPU 
resources:
    requests:
      memory: 128Mi
      cpu: 50m
    limits:
      memory: 128Mi
      cpu: 100m

however its only registry pod that we have configured with 2GB memory (1GB Registry container and 1GB for Registry Controller container).

resources:
      requests:
        memory: 1024Mi
        cpu: 100m
      limits:
        memory: 1024Mi
        cpu: 100m
